### PR TITLE
Improve XCM Docs

### PR DESCRIPTION
**Pull request by CrackTheCode016**
This PR is an effort to: 

- Improve, clarify, and restructure the existing XCM documentation
- Add more concrete examples of what XCM can do in different scenarios
- Cover what XCM v3 is adding to the protocol
- Add new XCM v3 instruction explanation
- ~~Compare XCM v2 vs XCM v3 to highlight the upgrade~~

### CODE REVIEW

**Comment by CrackTheCode016** [comment]
As a general note, this is just an initial PR - a more exhaustive one will be underway soon. This one makes sure everything is up to day and covers the latest in XCM, as well as improves some of what was already there.

File: docs/learn/learn-xcm.md [hunk 1]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
 ---
 
-Cross-Consensus Message Format (XCM) aims to be a language to communicate ideas between consensus
-systems. One of {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}'s main initiatives is
-that of interoperability, and XCM is the vehicle through which it will deliver this promise. Simply,
-it is a standard that allows protocol developers to define the data and origins which their chains
-can send and receive from. Out of the box, it comes with a VM that allows for customization of
-execution as well as the following properties:
+The Cross-Consensus Message Format, or **XCM**, is a **messaging format** and language used to
+define messages between consensus systems.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]: Does XCM define the messages or does it provide guidelines on how to construct them such that the communication can happen between different consensus systems? Just added an edit that made sense to me. Feel free to rephrase or improve on introducing it. I think the previous author took that sentence from here https://polkadot.network/blog/xcm-the-cross-consensus-message-format#-goals

```suggestion
The Cross-Consensus Message Format, or **XCM**, is a **messaging format** and language used to
communicate between consensus systems.
```

File: docs/learn/learn-xcm.md [hunk 2]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+
+One of {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}'s main initiatives is
+interoperability amongst parachains and any other participating consensus-driven systems. XCM is the
+language in which complex, cross-consensus conversations can occur. Two blockchains can "speak" XCM
+to seamlessly interact with each other using a standard messaging format.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
One of {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}'s main functionalities is
interoperability amongst parachains and any other participating consensus-driven systems. XCM is the
language through which complex, cross-consensus interactions can occur. Two blockchains can "speak" XCM
to seamlessly interact with each other using a standard messaging format.
```

File: docs/learn/learn-xcm.md [hunk 3]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+
+![xcm](../assets/cross-consensus/xcm.drawio.svg)
+
+:::info
+
+Remember, a consensus system here means any system or protocol that derives consensus to agree on
+the latest and correct state, whether it's a Polkadot parachain, an EVM smart contract, or other
+bridged consensus systems. We usually discuss XCM in the context of connecting parachains, but
+please bear this in mind!
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
We typically discuss XCM in the context of parachains, but please bear this in mind that it expands to the domain of all consensus systems!
Remember, a consensus system here means any system or protocol that achieves finality to agree on
the latest and correct state, whether it's a Polkadot parachain, an EVM smart contract, or other
bridged consensus systems. 
```

File: docs/learn/learn-xcm.md [hunk 4]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+
+XCM is not meant to be only specific to
+{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}, but rather its primary intention is
+to define a **generic** and **common** format amongst different consensus systems to communicate.
+
+It's important to note that XCM is not how messages are delivered but rather how they look, act, and
+contain relative instructions to whatever on-chain actions the message intends to perform.
+
+[**XCMP**](#xcmp-cross-chain-message-passing), or Cross Chain Message Passing, is the actual
+networking-layer protocol to deliver XCM-formatted messages to other participating parachains. There
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
network-layer protocol to deliver XCM-formatted messages to other participating parachains. There
```

File: docs/learn/learn-xcm.md [hunk 5]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+bridged consensus systems. We usually discuss XCM in the context of connecting parachains, but
+please bear this in mind!
+
+:::
+
+XCM is not meant to be only specific to
+{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }}, but rather its primary intention is
+to define a **generic** and **common** format amongst different consensus systems to communicate.
+
+It's important to note that XCM is not how messages are delivered but rather how they look, act, and
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
It's important to note that XCM does not define how messages are delivered but rather define how they should look, act, and
```

File: docs/learn/learn-xcm.md [hunk 6]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+contain relative instructions to whatever on-chain actions the message intends to perform.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
contain relative instructions to the on-chain actions the message intends to perform.
```

File: docs/learn/learn-xcm.md [hunk 7]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+
+[**XCMP**](#xcmp-cross-chain-message-passing), or Cross Chain Message Passing, is the actual
+networking-layer protocol to deliver XCM-formatted messages to other participating parachains. There
+are also other ways to define transport layer protocols for delivering XCM messages.
+
+XCM has four, high-level core design principles in which it stands to follow:
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
XCM has four high-level core design principles which it stands to follow:
```

File: docs/learn/learn-xcm.md [hunk 8]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
 2. **Absolute**: XCM messages are guaranteed to be delivered and interpreted accurately, in order
-   and in a timely fashion.
-3. **Asymmetric**: XCM messages out of the box do not have results that let the sender know that the
-   message was received. Any results must be separately communicated to the sender with an
-   additional message.
+   and in a timely fashion. Once a message is sent, one can be sure it will be processed as it was
+   intended to be.
+3. **Asymmetric**: XCM messages, by default, do not have results that let the sender know that the
+   message was received - they are 'fire and forget'. Any results must be separately communicated to
+   the sender with an additional message back to the origin.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
3. **Asymmetric**: XCM messages, by default, do not have results that let the sender know that the
   message was received - they follow the 'fire and forget' paradigm. Any results must be separately communicated to
   the sender with an additional message back to the origin.
```

File: docs/learn/learn-xcm.md [hunk 9]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
 4. **Agnostic**: XCM makes no assumptions about the nature of the Consensus Systems between which
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
4. **Agnostic**: XCM makes no assumptions about the nature of the consensus systems between which
```

File: docs/learn/learn-xcm.md [hunk 10]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
-   the messages are being passed.
+   the messages are being passed. XCM as a message format should be usable in any system that
+   derives consensus.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
   derives finality through consensus.
```

File: docs/learn/learn-xcm.md [hunk 11]
```diff
@@ -7,28 +7,60 @@ keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
...
+
+These four crucial design decisions allow for XCM messages to be a reliable, yet convenient way to
+properly convey the intentions from one consensus system to another without any compatibility
+limitations.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
These four crucial design decisions allow for XCM messages to be a reliable yet convenient way to
properly convey the intentions from one consensus system to another without any compatibility
issues.
```

File: docs/learn/learn-xcm.md [hunk 12]
```diff
@@ -54,6 +87,14 @@ not have reasonable interpretations under some systems or will be intentionally
 
 :::
 
+Furthermore, it's essential to realize that XCM themselves messages are _not_ transactions. XCM
+describes how to change the state of the target network, but the message itself doesn't perform the
+state change.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
Furthermore, it's essential to realize that XCM messages by themselves are _not_ considered transactions. XCM
describes how to change the state of the target network, but the message by itself doesn't perform the
state change.
```

File: docs/learn/learn-xcm.md [hunk 13]
```diff
@@ -54,6 +87,14 @@ not have reasonable interpretations under some systems or will be intentionally
...
+
+This partly ties what is called **asynchronous composability**, which allows XCM messages to bypass
+the concept of time-constrained mechanisms, like on-chain scheduling, and execute over time in the
+correct order in which it was intended.
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
This partly ties to what is called **asynchronous composability**, which allows XCM messages to bypass
the concept of time-constrained mechanisms, like on-chain scheduling and execution over time in the
correct order in which it was intended.
```

File: docs/learn/learn-xcm.md [hunk 14]
```diff
@@ -135,6 +176,95 @@ The destination deposits the derivative assets minted to the receiving account.
...
+:::info
+
+XCM v3's bridging, functional multichain decomposition, and programmability upgrades are crucial to
+bringing ecosystems together using a common communication abstraction.
+
+:::
+
+### XCM v3 Instruction Overview
+
+To reflect the functionality above, XCM v3 introduced many new instructions to include within its
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
To reflect the functionality above, XCM v3 introduced a multitude of new instructions to include within its
```

File: docs/learn/learn-xcm.md [hunk 15]
```diff
@@ -135,6 +176,95 @@ The destination deposits the derivative assets minted to the receiving account.
...
+messages. This list isn't exhaustive, however contains the necessary instruction sets to showcase
+the previously explained concepts of XCM v3.
+
+:::note
+
+Note most of these instruction definitions came from the source code, of which you may
+[explore for more depth!](https://github.com/paritytech/polkadot/blob/master/xcm/src/v3/mod.rs)
```
Comment (profradha) [inline docs/learn/learn-xcm.md]:
```suggestion
Note that most of these instruction definitions came from the source code, which you may
[explore for more depth!](https://github.com/paritytech/polkadot/blob/master/xcm/src/v3/mod.rs)
```

**Review by DrW3RK — COMMENTED** [review]
Some preliminary feedback improving readability

File: docs/learn/learn-xcm.md (hunk 1 above)
Comment (CrackTheCode016) [reply docs/learn/learn-xcm.md]: It technically does both, as it *defines* the format in which messages are constructed and executed. However in this context,  `communicate` works better 👍

**Review by DrW3RK — APPROVED** [review]

**Pull request merged**